### PR TITLE
Move become a supporter copy into subscribe link don't show to members

### PR DESCRIPTION
**Pull request by AWare**
## What does this change?

This adds the become a supporter link to the subscribe button, so that both may be shown. Previously, the become a supporter link was only shown to signed out users. This also hides the supporter link if the user is a paying member. I don't think we can do the same with the subscriber link presently. (I don't think the content authorisation proxy would handle the load)
## What is the value of this and can you measure success?

This should increase membership supporter sign ups without impacting subscriptions. 
## Screenshots

image
## Request for comment

@SiAdcock @samdesborough would appreciate comment on the js, and when we should handle hiding the membership link.
@tomverran for subs
@markjamesbutler for membership

### CODE REVIEW

**Comment by tomverran** [comment]
👍  though it would be nice if we could also show subscribe on tablet, since we sell a tablet edition

File: static/src/javascripts/projects/common/modules/navigation/membership.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+define(['common/modules/commercial/user-features', 'common/utils/fastdom-promise', 'Promise', 'common/utils/$'], function (userFeatures, fastdom, Promise, $) {
+    var LAST_CLASS = 'brand-bar__item--split--last';
+
+    function init() {
+        var refreshUser = new Promise(function (resolve) {
```
Comment (desbo) [inline static/src/javascripts/projects/common/modules/navigation/membership.js]: there's no need to assign this `Promise` to a var really, I'd return the entire chain from the function instead

File: static/src/javascripts/projects/common/modules/navigation/membership.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+            userFeatures.refresh();
+            resolve(userFeatures.isPayingMember());
```
Comment (desbo) [inline static/src/javascripts/projects/common/modules/navigation/membership.js]: it doesn't look like either of these `userFeatures` functions involve I/O; is there any need for them to be in the promise chain?

File: static/src/stylesheets/module/nav/_brand-bar.scss [hunk 3]
```diff
@@ -43,6 +43,24 @@
...
+    left: 11px;
+    position: relative;
+    height: 24px;
+    bottom: -5px;
+    display: block;
+}
+
+.brand-bar__item--split--last {
+    padding-left: 11px;
+    height: 34px;
```
Comment (SiAdcock) [inline static/src/stylesheets/module/nav/_brand-bar.scss]: Is it possible to use less magic numbers for these heights, positions and paddings? Perhaps some multiples of `$gs-gutter` (for horizontal spacing) `$gs-baseline` (for vertical spacing) and `$gs-height` (for element height)

File: static/src/javascripts/projects/common/modules/navigation/membership.js (hunk 2 above)
Comment (AWare) [inline static/src/javascripts/projects/common/modules/navigation/membership.js]: refresh does some async stuff, but not in a way that's useful here, I'm not going to call refresh, and just not show the supporter link if we don't know membership status

**Comment by SiAdcock** [comment]
Very nice, man 👍

**Comment by desbo** [comment]
ANOTHER ONE 👍

**Pull request merged**